web/src/app/pages/main-page-container/main-page/map/map.component.ts

```diff
@@ -434,46 +448,51 @@ export class MapComponent implements AfterViewInit, OnDestroy {
       if (this.shouldEnableDrawingTools) {
         marker.setDraggable(true);
       }
+      this.panAndZoom(marker?.getPosition());
     }
+  }
+
+  private unselectMarker() {
     const selectedMarker = this.selectedLocationOfInterestId
       ? this.markers.get(this.selectedLocationOfInterestId)
       : undefined;
+
     if (selectedMarker) {
       this.setIconSize(selectedMarker, normalIconScale);
       selectedMarker.setDraggable(false);
     }
-    this.panAndZoom(marker?.getPosition());
-  }
-
-  private setIconSize(marker: google.maps.Marker, size: number) {
-    const icon = marker.getIcon() as google.maps.Icon;
-    const newIcon = {
-      url: icon.url,
-      scaledSize: {
-        width: size,
-        height: size,
-      },
-    } as google.maps.Icon;
-    marker.setIcon(newIcon);
   }
 
   private selectPolygons(locationOfInterestId: string | null) {
     const polygons = locationOfInterestId
       ? this.polygons.get(locationOfInterestId)
       : undefined;
-    if (polygons) {
-      for (const polygon of polygons) {
-        polygon.setOptions({strokeWeight: enlargedPolygonStrokeWeight});
-      }
-    }
+
+    const bounds = new google.maps.LatLngBounds();
+
+    polygons?.forEach(polygon => {
+      polygon.setOptions({strokeWeight: enlargedPolygonStrokeWeight});
+
+      polygon.getPath().forEach(path => {
+        const point = new google.maps.LatLng(path.lat(), path.lng());
+        bounds.extend(point);
+      });
+
+      this.map.googleMap?.fitBounds(bounds);
+    });
+  }
+
+  private unselectPolygons() {
```

Comment:
This code can be simplified:

if (this.selectedLoiIds) {
  Do stuff
}

Without the initial assignment.